crates/router/src/connector/adyen.rs

```diff
@@ -720,11 +721,13 @@ impl api::IncomingWebhook for Adyen {
     fn get_webhook_object_reference_id(
         &self,
         request: &api::IncomingWebhookRequestDetails<'_>,
-    ) -> CustomResult<String, errors::ConnectorError> {
+    ) -> CustomResult<ObjectReferenceId, errors::ConnectorError> {
         let notif = get_webhook_object_from_body(request.body)
             .change_context(errors::ConnectorError::WebhookReferenceIdNotFound)?;
 
-        Ok(notif.psp_reference)
+        Ok(ObjectReferenceId::PaymentId(
+            api_models::payments::PaymentIdType::ConnectorTransactionId(notif.psp_reference),
+        ))
```

Comment:
Need logic to detect type of webhook and return appropriate ObjectReferenceId as opposed to always returning PaymentId

Reply:
Done